src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp

```diff
@@ -1766,6 +1766,10 @@ int xcldev::xclReset(int argc, char *argv[])
 
     std::string vbnv, errmsg;
     auto dev = pcidev::get_dev(index);
+    if (!dev->is_ready) {
```

Comment:
should we also check if device is offline? offline means a previous reset has not finished or driver has not finished probing, so not ready for handling reset?

Reply:
If the previous reset has issue or driver probe failed, then device is not in ready state. So, I think one more "dev_offline" check may not be required. Please let me know if this is not the case, I will add dev_offline check as well here.

Reply:
@rajkumar-xilinx , thanks for clarifying. do we know why device is not in ready state if the previous reset has failed? I'm wondering if we should also add dev_offline as one of the condition in ready sysfs node? Should we consider device as ready if it's not online, @houlz0507 @xuhz ?

Reply:
seems the device is always online when it is ready. again, offline caused by a failed reset is a regression.

Reply:
The condition we check for ready does not include online/offline today. The 'ready' sysfs node is created before device is set to online (which is the very last step in probe() routine). So there is a small window when the 'ready' sysfs node says 1, but device is not online yet. @houlz0507 can correct me, if I'm wrong.